src/item.h

```diff
@@ -354,7 +356,7 @@ class item : public JsonSerializer, public JsonDeserializer, public visitable<it
     std::string info( bool showtext, std::vector<iteminfo> &dump ) const;
 
     /** Burns the item. Returns true if the item was destroyed. */
-    bool burn( fire_data &bd );
+    bool burn( fire_data &bd, bool can_spread = false );
```

Comment:
Just supply the correct argument there, don't default. `burn(thing, false)`

Though you also should change the argument from `can_spread` to something like `contained`. It would make it more informative and also cover the player burning example better.